dogfood/main.tf

```
@@ -56,6 +56,10 @@ resource "coder_agent" "dev" {
     if [ -n "$DOTFILES_URI" ]; then
       coder dotfiles "$DOTFILES_URI" -y 2>&1 | tee  ~/.personalize.log
     fi
+
+    # Install Nix into our bash profile so `nix-shell`, `nix-build, and `nix` are available
+    echo '. /home/coder/.nix-profile/etc/profile.d/nix.sh' >> /home/coder/.bashrc
+    bash /opt/nix/install --no-daemon
```

Didn't test this, but imagine it should be a bit more portable and avoid adding the entry every startup.

```suggestion
    bash /opt/nix/install --no-daemon
    if ! grep -q '. ~/.nix-profile/etc/profile.d/nix.sh' ~/.bashrc; then
      echo '. ~/.nix-profile/etc/profile.d/nix.sh' >> ~/.bashrc
    fi
```

Ultimately, we might want to add this logic into `/etc/profile.d/nix.sh` and `/etc/zsh/zprofile` (via Dockerfile) so that it's available in all shells. But that can be done in a separate PR.